### ISSUE TITLE
Service Discovery can't be disabled

### ISSUE THREAD

**Issue by m-kratochvil** [F5Networks/f5-appsvcs-extension#502]
### Environment
 * Application Services Version:
```
  "version": "3.30.0",
  "release": "5",
  "schemaCurrent": "3.30.0",
  "schemaMinimum": "3.0.0"
```

 * BIG-IP Version:
```
Sys::Version
Main Package
  Product     BIG-IP
  Version     15.1.3
  Build       0.21.11
  Edition     Engineering Hotfix
  Date        Wed Apr 28 09:38:30 PDT 2021
```

### Summary
I want to disable the Service Discovery.
As described in the AS3 documentation [HERE](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/declarations/discovery.html#disabling-service-discovery) I am issuing following local API call which returns an error:
```
restcurl -u admin:XXXXX -X POST -d '{"serviceDiscoveryEnabled": false}' /mgmt/shared/appsvcs/settings                          {
  "code": 500,
  "message": "GET http://admin:XXXXXX@localhost:8100/mgmt/shared/service-discovery/task getting Service Discovery tasks response\u003d404 body\u003d{\"code\":404,\"message\":\"\",\"referer\":\"Unknown\",\"errorStack\":[]}"
}
```

Further confirmed by checking the Service Discovery `task` endpoint:
```
 restcurl -u admin:XXXXX /mgmt/shared/service-discovery/task
{
  "code": 404,
  "message": "",
  "referer": "Unknown",
  "errorStack": []
}
```

And the AS3 `settings` endpoint keep showing Service Discovery is enabled:
```
restcurl -u admin:XXXXX /mgmt/shared/appsvcs/settings
{
  "burstHandlingEnabled": false,
  "asyncTaskStorage": "memory",
  "serviceDiscoveryEnabled": true
}
```

### Steps To Reproduce
As above

### Expected Behavior
Service Discovery should be disabled and I would expect the output of `settings` endpoint to look like this:
```
restcurl -u admin:XXXXX /mgmt/shared/appsvcs/settings
{
  "burstHandlingEnabled": false,
  "asyncTaskStorage": "memory",
  "serviceDiscoveryEnabled": false
}
```

In the case of the `task` endpoint returning HTTP 404 error, I think it would make sense that Service Discovery should simply be disabled instead of erroring out.

### Actual Behavior
As shown above, Service Discovery remains enabled because the `task` endpoint fails the query (returns HTTP 404 in my case).

**Comment by dstokesf5** [comment]
Thank you for reporting this. I have added it to our internal product backlog as AUTOTOOL-2719.

**Comment by j-zehnder** [comment]
I just ran into this issue as well, looking forward to an update.

**Comment by walterk82** [comment]
When looking in the filesystem at /var/config/rest/iapps I have noticed that f5-service-discovery is not present for the boxes that exhibit this condition.  It works in all other cases where this directory is present.  I believe that the installation is corrupt in this case.  I have filed RFE 995173 to ask that service-discovery installation process be revisited.  Should RFE 995173 be implemented then this end-point would no longer be necessary.

**Comment by m-kratochvil** [comment]
Just upgraded AS3 to 3.22 and ServiceDiscovery can be correctly disabled now.

```
restcurl -u admin:XXXXX -X POST -d '{"serviceDiscoveryEnabled": false}' /mgmt/shared/appsvcs/settings                              {
  "asyncTaskStorage": "data-group",
  "burstHandlingEnabled": false,
  "performanceTracingEnabled": false,
  "performanceTracingEndpoint": "",
  "serviceDiscoveryEnabled": false
}
```

I did notice a new error message in `/var/log/restnoded/restnoded.log`:
```
severe: [RestWorker] [RestWorker] worker failed to start: TypeError: Cannot convert undefined or null to object:shared/service-discovery/task
```

Not sure if this is causing any impact, but it does show during every `restnoded` daemon restart.

I also found another issue with the AS3 settings, if interested, see #525

**Comment by dstokesf5** [comment]
This has been resolved in the 3.32.0 release.
